File: cantor.py
import matplotlib.pyplot as plt

# рекурсия для построения множества 
def drawCantorSet(x, y, length, depth):
    if depth == 0:
        return

    plt.plot([x, x + length], [y, y], color='black', linewidth=2)

    # рекурсия для левой и правой части
    newY = y - 20  # расстояние между уровнями 
    newLength = length / 3.0
    drawCantorSet(x, newY, newLength, depth - 1)  # левая часть
    drawCantorSet(x + 2 * newLength, newY, newLength, depth - 1)  # правая часть

def main():
    width = 2000
    height = 600
    depth = 7  # количество уровней множества

    plt.figure(figsize=(width / 100, height / 100))  # размер окна
    plt.axis('off')
    drawCantorSet(50, height - 50, 1800, depth)  # начальная точка и длина
    plt.show()

if __name__ == "__main__":
    main()

File: cantor.cpp
#include <SFML/Graphics.hpp>

// рекурсия для построения множества кантора
void drawCantorSet(sf::RenderWindow& window, float x, float y, float length, int depth) {
    if (depth == 0) return;

    // отрезок
    sf::RectangleShape line(sf::Vector2f(length, 2)); // высота 2 пикселя
    line.setPosition(x, y);
    line.setFillColor(sf::Color::Black);
    window.draw(line);

    // рекурсия для левой и правой части отрезка
    float newY = y + 20; // расстояние между уровнями
    float newLength = length / 3.0f;
    drawCantorSet(window, x, newY, newLength, depth - 1);                 // левая часть
    drawCantorSet(window, x + 2 * newLength, newY, newLength, depth - 1); // правая часть
}

int main() {
    const int width = 2000;
    const int height = 600;
    const int depth = 6; // количество уровней множества

    sf::RenderWindow window(sf::VideoMode(width, height), "Cantor Set");

    while (window.isOpen()) {
        sf::Event event;
        while (window.pollEvent(event)) {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear(sf::Color::White); // фон
        drawCantorSet(window, 50, 20, 1800, depth); // начальная точка и длина
        window.display();
    }

    return 0;
}
